### PR TITLE
refactor: discounted fee

### PR DESCRIPTION
**Pull request by Roznovjak**
Make the discounted fee configurable instead of using hardcoded value.

### CODE REVIEW

**Comment by lumir-mrkva** [comment]
Task linked: [CU-2jfyk51 XYK - refactor hardcoded discounted fee](https://app.clickup.com/t/2jfyk51)

**Review by enthusiastmartin — CHANGES_REQUESTED** [review]
I'd suggest to add few more tests with different fees (zero included).

Now it is set to the original default value 0.07% but it is not really tested whether it works with different values.

**Comment by Roznovjak** [comment]
> I'd suggest to add few more tests with different fees (zero included).
> 
> Now it is set to the original default value 0.07% but it is not really tested whether it works with different values.

We don't test config types for various values because it would require a lot of mocked runtimes.
But the `DiscountedFee` is used only in one method, and this method is properly [tested](https://github.com/galacticcouncil/HydraDX-math/blob/33c7749147fa5766f4e1cbf2019bfa654d3683aa/src/fee.rs#L90).

**Comment by enthusiastmartin** [comment]
> > I'd suggest to add few more tests with different fees (zero included).
> > Now it is set to the original default value 0.07% but it is not really tested whether it works with different values.
> 
> We don't test config types for various values because it would require a lot of mocked runtimes. But the `DiscountedFee` is used only in one method, and this method is properly [tested](https://github.com/galacticcouncil/HydraDX-math/blob/33c7749147fa5766f4e1cbf2019bfa654d3683aa/src/fee.rs#L90).

you can test similar way how exchange fee is tested for xyk pallet. 

you can easily build tested runtime with different fee.

File: pallets/xyk/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = 'pallet-xyk'
-version = '3.6.5'
+version = '3.7.0'
```
Comment (enthusiastmartin) [inline pallets/xyk/Cargo.toml:3]: i think this should be a major version change . 

Because it introduces a change which is non backward compatible.wdyt ?

Comment (Roznovjak) [reply pallets/xyk/Cargo.toml:3]: Not sure if a pallet Config belongs to public API. But we can bump the major version.

Comment (enthusiastmartin) [reply pallets/xyk/Cargo.toml]: sure it does. you can upgrade pallet-xyk version in runtime as the config changes.
